src/ClientData/CaptureData.cpp

```diff
@@ -91,7 +91,10 @@ void CaptureData::AddScopeStats(ScopeId scope_id, ScopeStats stats) {
   all_scopes_->SetScopeStats(scope_id, stats);
 }
 
-void CaptureData::OnCaptureComplete() { thread_track_data_provider_->OnCaptureComplete(); }
+void CaptureData::OnCaptureComplete() {
+  thread_track_data_provider_->OnCaptureComplete();
+  all_scopes_->SortTimers();
```

Comment:
`CaptureData` doesn't need to know that the scopes need to be sorted post-capturing. I would have a `OnCaptureComplete()` method in `all_scopes_` that simply call `SortTimers().`

Too much bureaucracy?

Reply:
As long as it's the only spot where `SortTimers()` is called outside of ScopeCollection, I support the suggestion.

Reply:
Fair; changed to OnDataChanged (valeriy also suggested above) since ScopeStatsCollection also doesn't need to know that the capture has completed, but simply that it's finished changing the data (especially as this gets called even on a loaded capture).